pages/positions/openloan.tsx

```diff
@@ -1,10 +1,295 @@
+      setButtonText('Tokens must be different')
+      setConfirmStyle(style.confirmGrey)
+      return false
+    }
+    if (isTokenEmpty(token1)) {
+      setButtonText('Token must be selected')
+      setConfirmStyle(style.confirmGrey)
+      return false
+    }
+    if (collateralType == CollateralType.None) {
+      setButtonText('Collateral must be selected')
+      setConfirmStyle(style.confirmGrey)
+      return false
+    }
+    if (loanAmt <= 0) {
+      setButtonText('Loan amount must be positive')
+      setConfirmStyle(style.confirmGrey)
+      return false
+    }
+    if (collateralAmt0 <= 0) {
+      setButtonText(token0.symbol + ' collateral amount must be positive')
+      setConfirmStyle(style.confirmGrey)
+      return false
+    }
+    if (collateralType == CollateralType.Both && collateralAmt1 <= 0) {
+      setButtonText(token1.symbol + ' collateral amount must be positive')
+      setConfirmStyle(style.confirmGrey)
+      return false
+    }
+    setButtonText('Confirm')
+    setConfirmStyle(style.confirmGreen)
+    return true
+  }
+
+  function isTokenEmpty(tokenToCheck: Token): boolean {
+    return Object.values(tokenToCheck).every((tokenProp) => tokenProp === '')
+  }
+
+  async function validateBeforeSubmit(data: FieldValues): Promise<void> {
+    if (!validate()) {
+      return
+    }
+    return openLoanHandler(data)
+  }
+
+  // checks for non-numeric value inputs
+  const validateNumberInput = (e: ChangeEvent<HTMLInputElement> | string, setNumberInputStr: Dispatch<SetStateAction<string>>, setNumberInputval: Dispatch<SetStateAction<number>>): void => {
+    let numberInputStr: string
+    if (typeof e !== 'string') numberInputStr = (e.target as HTMLInputElement).value
+    else numberInputStr = e
+
+    var strToSet = ''
+    var i = numberInputStr.indexOf('.')
+    if (i >= 0 && i + 1 < numberInputStr.length) {
+      strToSet = numberInputStr.substring(0, i + 1) + numberInputStr.substring(i + 1).replace(/[^0-9]/g, '')
+    } else {
+      strToSet = numberInputStr.replace(/[^0-9\.]/g, '')
+    }
+    setNumberInputStr(strToSet)
+
+    // clamp the value
+    var inputVal = parseFloat(strToSet)
+    if (!isNaN(inputVal)) {
+      setNumberInputval(inputVal)
+    }
+  }
+
+  const handleNumberInput = (e: ChangeEvent<HTMLInputElement> | string, setNumberInputStr: Dispatch<SetStateAction<string>>, setNumberInputval: Dispatch<SetStateAction<number>>) => {
+    try {
+      if (e) {
+        const numberInput = typeof e !== 'string' ? e.target.value : e
+        if (numberInput === '') {
+          setNumberInputStr('')
+        } else {
+          validateNumberInput(numberInput, setNumberInputStr, setNumberInputval)
+        }
+      }
+      validate()
+    } catch (error) {
+      let message
+      if (error instanceof Error) message = error.message
+      else message = String(error)
+
+      notifyError(message)
+    }
+  }
+
+  setValue('collateralType', collateralType)
+  return (
+    <>
```

Comment:
is this needed? I feel like you can remove it and it'll still be fine.

Reply:
In react a component must return only single element. By using the empty tag syntax one doesn’t have to either import React or write the long keyword React.Fragment every time. So it’s basically a Syntactic sugar which makes our lives easier.

Reply:
Done